src/containers/MainPlotContainer/index.tsx

```diff
@@ -173,13 +174,17 @@ class MainPlotContainer extends React.Component<MainPlotContainerProps> {
 
     public renderPopover() {
         const { hoveredPointData, galleryCollapsed, thumbnailRoot } = this.props;
+        const thumbnailSrc = formatThumbnailSrc(
+            thumbnailRoot,
+            hoveredPointData?.thumbnailPath || ""
+        );
```

Comment:
could we do this earlier? like in the image-dataset class functions?

Reply:
Probably, but I think I'd need to change functionality in the existing ImageDataset classes as well, and I didn't want to touch them too much in this PR.